docs/current_docs/manuals/developer/entrypoint-function.mdx

```diff
@@ -0,0 +1,168 @@
+
+## Simple entrypoint function
+
+The entrypoint function for a module can be overridden by registering a custom constructor. Its parameters are available as flags in the `dagger call` command directly.
+
+:::note
+Dagger modules have only one entrypoint function. Entrypoint function of [custom types](./custom-types.mdx) are not registered; they are constructed by the function that [chains](./chaining.mdx) them.
+:::
+
+Here is an example module with a custom constructor as entrypoint function:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/entrypoint/go/simple/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/entrypoint/python/simple/main.py
+```
+
+:::info
+In the Python SDK, the [`@dagger.object_type`](https://dagger-io.readthedocs.io/en/latest/module.html#dagger.object_type) decorator wraps [`@dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html), which means that an `__init__()` method is automatically generated, with parameters that match the declared class attributes.
+:::
+
+The code listing above is an example of an object that has typed attributes.
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./snippets/entrypoint/typescript/simple/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Here is an example call for this module:
+
+```shell
+dagger call --name=Foo message
+```
+
+The result will be:
+
+```shell
+Hello, Foo!
+```
+
+:::note
+If you plan to use constructor fields in other module functions, ensure that they are declared as public (in Go and TypeScript). This is because Dagger stores fields using serialization and private fields are omitted during the serialization process. As a result, if a field is not declared as public, calling methods that use it will produce unexpected results.
+:::
+
+## Exclude argument in constructor (Python)
+
+:::important
+The information in this section is only applicable to the Python SDK.
+:::
+
+Same as any [data class](https://docs.python.org/3/library/dataclasses.html), attributes can be excluded from the
+generated `__init__()` function, using [dataclasses.field(init=False)](https://docs.python.org/3/library/dataclasses.html#dataclasses.field):
+
+```python file=./snippets/entrypoint/python/exclude/main.py
+```
+
+In this case, only the `name` flag was added and is visible in the output:
+
+```shell
+FUNCTIONS
+  message       Return the greeting message
+
+ARGUMENTS
+      --name string   Who to greet (default "World")
+```
+
+## Constructor-only arguments (Python)
+
+:::important
+The information in this section is only applicable to the Python SDK.
+:::
+
+The opposite is also possible. To define an argument that only exists in
+the constructor, but not as a class attribute, define it as an
+[init-only variable](https://docs.python.org/3/library/dataclasses.html#init-only-variables):
+
+```python file=./snippets/entrypoint/python/initvar/main.py
+```
+
+## Complex or mutable defaults (Python)
```

Comment:
You can call `new` to constructor a complex object or `dag.xxx` for a dagger object inside the constructor